### ISSUE TITLE
Memory leak in TabControlWmReflectDrawItem(), dispose not called on graphics

### ISSUE THREAD

**Issue by jgilbert2017** [dotnet/winforms#10534]
### .NET version

8.0

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

unknown

### Issue description

See code below.  The `Graphics` object created by `dis->hDC.CreateGraphics()` is not disposed (neither via `DrawItemEventArgs.Dispose()` nor `WmReflectDrawItem()`).  This results in the object having its finalizer run.

https://github.com/dotnet/winforms/blob/9e2fc95fca4e6d18444c819b35bb1338ac6ca63d/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TabControl/TabControl.cs#L1935

```
    private unsafe void WmReflectDrawItem(ref Message m)
    {
        DRAWITEMSTRUCT* dis = (DRAWITEMSTRUCT*)(nint)m.LParamInternal;

        using DrawItemEventArgs e = new(
            dis->hDC.CreateGraphics(),
            Font,
            dis->rcItem,
            (int)dis->itemID,
            (DrawItemState)(int)dis->itemState);

        OnDrawItem(e);

        m.ResultInternal = (LRESULT)1;
    }
```

### Steps to reproduce

n/a

**Comment by kirsan31** [comment]
Good catch @jgilbert2017, but this is not memory leak - this is performance issue.
The fix is trivial - do you planning make a PR?

**Comment by jgilbert2017** [comment]
@kirsan31 if making a PR gets it fixed quicker i'd be happy to.  is this how you want the code to look?

```
    private unsafe void WmReflectDrawItem(ref Message m)
    {
        DRAWITEMSTRUCT* dis = (DRAWITEMSTRUCT*)(nint)m.LParamInternal;

        using var graphics = dis->hDC.CreateGraphics();
        using DrawItemEventArgs e = new(
            graphics,
            Font,
            dis->rcItem,
            (int)dis->itemID,
            (DrawItemState)(int)dis->itemState);

        OnDrawItem(e);

        m.ResultInternal = (LRESULT)1;
    }
```

**Comment by kirsan31** [comment]
> @kirsan31 if making a PR gets it fixed quicker i'd be happy to. is this how you want the code to look?
> 
> ```cs
>     private unsafe void WmReflectDrawItem(ref Message m)
>     {
>         DRAWITEMSTRUCT* dis = (DRAWITEMSTRUCT*)(nint)m.LParamInternal;
> 
>         using var graphics = dis->hDC.CreateGraphics();
>         using DrawItemEventArgs e = new(
>             graphics,
>             Font,
>             dis->rcItem,
>             (int)dis->itemID,
>             (DrawItemState)(int)dis->itemState);
> 
>         OnDrawItem(e);
> 
>         m.ResultInternal = (LRESULT)1;
>     }
> ```

That's exactly what I would do.

**Comment by Tanya-Solyanik** [comment]
Related changes [PR3553](https://github.com/dotnet/winforms/commit/a6a308fa452770caafca55ee4d8ac54d1ee1015d#diff-edd2fb85653c19f4c380ec1ae98b35d6723ca3ecb80ef29a4be708f83fa30463) and [PR9309](https://github.com/dotnet/winforms/commit/e9fc1dac4f9792856f0bc8932486303dd8a8fe02#diff-edd2fb85653c19f4c380ec1ae98b35d6723ca3ecb80ef29a4be708f83fa30463)

**Comment by kirsan31** [comment]
@Tanya-Solyanik 
So, the regression was introduced in https://github.com/dotnet/winforms/pull/9309
Quick search by `CreateGraphics()` on this PR bring that we have same problem in [ComboBox](https://github.com/dotnet/winforms/commit/e9fc1dac4f9792856f0bc8932486303dd8a8fe02#diff-a279bfceaf830444fc4318432b3387481fe38b4b5fdde61a216bdd3bf8a944feR3750) and [ListBox](https://github.com/dotnet/winforms/commit/e9fc1dac4f9792856f0bc8932486303dd8a8fe02#diff-3fc9c1f5b9b56f994ae2dce4b892c52742649d8c754470c83cb951b10da00ddeR2413) too.

May be @jgilbert2017 will fix them too?

**Comment by jgilbert2017** [comment]
@kirsan31 somehow the `main` branch doesn't seem to have the issue for `ComboBox` and `ListBox` although i did come across of `Dispose()` being used manually w/o a `try/finally` or `using` statement.

**Comment by kirsan31** [comment]
> somehow the `main` branch doesn't seem to have the issue for `ComboBox` and `ListBox`

Yea, they was changed later to other overload (that create internal graphics, which is disposed):
https://github.com/dotnet/winforms/blob/c26044534271f1eb06e1487b0ecb8027c1e4279e/src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.cs#L3763-L3770

**Comment by elachlan** [comment]
We should probably pass in HDC instead of calling create graphics then?

**Comment by kirsan31** [comment]
> We should probably pass in HDC instead of calling create graphics then?

So that everything is in the same style - of course.
